metadata-ingestion/src/datahub/ingestion/source/sql/sql_generic_profiler.py

```diff
@@ -118,15 +125,34 @@ def is_dataset_eligible_for_profiling(
         size_in_bytes: Optional[int],
         rows_count: Optional[int],
     ) -> bool:
-        threshold_time: Optional[datetime] = None
+        dataset_urn = make_dataset_urn_with_platform_instance(
+            self.platform,
+            dataset_name,
+            self.config.platform_instance,
+            self.config.env,
+        )
+
+        if not self.config.table_pattern.allowed(dataset_name):
+            return False
+
+        last_profiled: Optional[int] = None
+        if self.state_handler:
+            last_profiled = self.state_handler.get_last_profiled(dataset_urn)
+            if last_profiled:
+                # If profiling state exists we have to carry over to the new state
+                self.state_handler.add_to_state(dataset_urn, last_profiled)
+
+        threshold_time: Optional[datetime] = (
+            datetime.fromtimestamp(last_profiled / 1000, timezone.utc)
+            if last_profiled
+            else None
+        )
         if self.config.profiling.profile_if_updated_since_days is not None:
             threshold_time = datetime.now(timezone.utc) - timedelta(
```

Comment:
I think, if threshold_time (last profiling time for a table) is available in profiling state, it would be smarter not to override it by threshold_time computed from `profile_if_updated_since_days`.

Reply:
thanks, I will address this